networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1133 @@
+
+    Return
+    -------
+    entropy: float
+        graph entropy, in [1]_, eq.14/16 ( H_b/f (G_C) )
+
+    Examples
+    ---------
+    >>> import numpy as np
+    >>> b = np.array([
+    ...     [0, 0, 1, 0, 0, 0, 1],
+    ...     [0, 0, 1, 0, 0, 0, 0],
+    ...     [0, 0, 0, 0, 1, 0, 0],
+    ...     [0, 0, 1, 0, 1, 0, 0],
+    ...     [0, 0, 0, 0, 0, 1, 0],
+    ...     [0, 0, 0, 1, 0, 0, 1],
+    ...     [0, 0, 0, 0, 0, 0, 0],
+    ... ])
+
+    >>> condensed_network_layers = nx.recursive_leaf_removal(nx.condensation(nx.from_numpy_matrix(b, create_using=nx.DiGraph)))
+    >>> fwd_graph_entropy = [round(nx.graph_entropy(net, forward_entropy=True), 3) for net in condensed_network_layers]
+    >>> bkwd_graph_entropy = [round(nx.graph_entropy(net), 3) for net in condensed_network_layers]
+    >>> print("graph entropy (from top | bottom): {0} | {1}".format(fwd_graph_entropy, bkwd_graph_entropy))
+    graph entropy (from top | bottom): [0.52, 0.0] | [1.04, 0.0]
+
+    Notes
+    ______
+    WIP: Detail after confirmation of methodology from Prof. Sole.
+
+    As described in the text of _[1], proposed and developed in _[2], _[3]:
+    Graph Entropy, is measure of probability of crossing a node n_j when following a path from n_i.
+
+    _Forward_ entropy:
+    .. math:: H_f(G_C) = \frac{1}{|M|} \sum_{v_i \in M} \sum_{v_k \in V \setminus \mu} P(v_i \right v_k) \cdot log k_{out}(v_k)
+    _Backward_ entropy:
+    .. math:: H_b(G_C) = \frac{1}{|\mu|} \sum_{v_i \in \mu} \sum_{v_k \in V_C \setminus M} P(v_i \leftarrow v_k) \cdot log k_{in}(v_k)
+    where :math:`\mu` (M) is the set of minimal (maximal) nodes, and :math:`P(v_i \leftarrow v_k)` indicates
+    the probability of transition between :math:`v_k` and :math:`v_i`.
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    .. [2] "Topological reversibility and causality in feed-forward networks."
+      Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquín Goñi, and Ricard Solé.
+      New Journal of Physics 12, no. 11 (2010): pg 113051
+
+    .. [3] "Measuring the hierarchy of feedforward networks."
+    Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquin Goni, and Ricard Solé.
+    Chaos: An Interdisciplinary Journal of Nonlinear Science 21, no. 1 (2011): pg 016108.
+
+    """
+    import numpy as np
+
+    if not nx.is_directed_acyclic_graph(DAG):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for graph entropy evaluation"
+        )
+
+    dag = nx.convert_node_labels_to_integers(DAG)
+    # Could be passed in most contexts to reduce redundant computation
+    L_GC = len(recursive_leaf_removal(DAG))
+
+    axis = 1 if forward_entropy else 0
+    B = _matrix_normalize(nx.to_numpy_array(dag), axis=axis)
```

Comment:
```suggestion
    B = nx.to_numpy_array(dag)
    # normalize so rows (forward_entropy) or columns (backward_entropy) sum to 1. 
    sums = B.sum(axis=axis, keepdims=True)
    sums[sums==0] = 1
    B = B / sums
```
This is all I was thinking of for "inlining"...

Reply:
done!